Add support for certificates in PKCS#12 (P12) key stores (#53810)
elastic/kibana: docs/setup/settings.asciidoc

`elasticsearch.sniffOnConnectionFault:`:: *Default: false* Update the list of
Elasticsearch nodes immediately following a connection fault.

`elasticsearch.ssl.alwaysPresentCertificate:`:: *Default: false* Controls whether to always present the certificate specified by

> `elasticsearch.ssl.certificate` or `elasticsearch.ssl.keystore.path` when requested. This applies to all requests to Elasticsearch,

Reviewer: Sometimes it is unclear what "this" applies to in some of these descriptions. In this case, I think it is referring to the setting?
```suggestion
`elasticsearch.ssl.certificate` or `elasticsearch.ssl.keystore.path` when requested. This setting applies to all requests to Elasticsearch,
```

> including requests that are proxied for end-users. Setting this to `true` when Elasticsearch is using certificates to authenticate users can

Reviewer: ```suggestion
including requests that are proxied for end users. Setting this to `true` when Elasticsearch is using certificates to authenticate users can
```

> lead to proxied requests for end-users being executed as the identity tied to the configured certificate.

Reviewer: ```suggestion
lead to proxied requests for end users being executed as the identity tied to the configured certificate.
```

`elasticsearch.ssl.certificate:` and `elasticsearch.ssl.key:`:: Paths to a PEM-encoded X.509 certificate and its private key, respectively.
When `xpack.security.http.ssl.client_authentication` in Elasticsearch is set to `required`, the certificate and key are used to prove
Kibana's identity when it makes an outbound request to your Elasticsearch cluster.

> 

Reviewer: Does this note apply to the elasticsearch.ssl.certificate and elasticsearch.ssl.key settings?  If so, I think it should be linked to that description.  For example:
```suggestion
+
--
```

> NOTE: These settings cannot be used in conjunction with `elasticsearch.ssl.keystore.path`.

Reviewer: Per above, if this NOTE is applicable to the previous definition, it should be grouped as follows:
```suggestion
NOTE: These settings cannot be used in conjunction with `elasticsearch.ssl.keystore.path`.

--
```

> `elasticsearch.ssl.certificateAuthorities:`:: Paths to one or more PEM-encoded X.509 certificates. This may consist of a root certificate

Reviewer: Another slightly unclear "this":
```suggestion
`elasticsearch.ssl.certificateAuthorities:`:: Paths to one or more PEM-encoded X.509 certificates. These certificates may consist of a root certificate
```

> authority (CA), and one or more intermediate CAs, which make up a trusted certificate chain for Kibana. This is used to establish trust when

Reviewer: ```suggestion
authority (CA) and one or more intermediate CAs, which make up a trusted certificate chain for Kibana. This is used to establish trust when
```

Kibana creates an SSL connection with your Elasticsearch cluster. In addition to this setting, trusted certificates may be specified via
`elasticsearch.ssl.keystore.path` and/or `elasticsearch.ssl.truststore.path`.

`elasticsearch.ssl.keyPassphrase:`:: The passphrase that will be used to decrypt the private key that is specified via
`elasticsearch.ssl.key`. This value is optional, as the key may not be encrypted.

> `elasticsearch.ssl.keystore.path:`:: Path to a PKCS #12 file -- this should contain an X.509 certificate with its private key. When

Reviewer: ```suggestion
`elasticsearch.ssl.keystore.path:`:: Path to a PKCS #12 file that contains an X.509 certificate with its private key. When
```

`xpack.security.http.ssl.client_authentication` in Elasticsearch is set to `required`, the certificate and key are used to prove Kibana's
identity when it makes an outbound request to your Elasticsearch cluster. If the file contains any additional certificates, those will be

> used as a trusted certificate chain for your Elasticsearch cluster. This is used to establish trust when Kibana creates an SSL connection

Reviewer: Qualifying another "this"
```suggestion
used as a trusted certificate chain for your Elasticsearch cluster. This chain is used to establish trust when Kibana creates an SSL connection
```

with your Elasticsearch cluster. In addition to this setting, trusted certificates may be specified via
`elasticsearch.ssl.certificateAuthorities` and/or `elasticsearch.ssl.truststore.path`.

> 

Reviewer: To have this NOTE show up properly indented, you need to tie it to the previous paragarph:
```suggestion
+
--
```

> NOTE: This setting cannot be used in conjunction with `elasticsearch.ssl.certificate` or `elasticsearch.ssl.key`.

Reviewer: Closing the block:
```suggestion
NOTE: This setting cannot be used in conjunction with `elasticsearch.ssl.certificate` or `elasticsearch.ssl.key`.

--
```

`elasticsearch.ssl.keystore.password:`:: The password that will be used to decrypt the key store and its private key. If your key store has
no password, leave this unset. If your key store has an empty password, set this to `""`.

> `elasticsearch.ssl.truststore.path:`:: Path to a PKCS #12 trust store -- this should contain one or more X.509 certificates. This may

Reviewer: ```suggestion
`elasticsearch.ssl.truststore.path:`:: Path to a PKCS #12 trust store that contains one or more X.509 certificates. This may
```

> consist of a root certificate authority (CA), and one or more intermediate CAs, which make up a trusted certificate chain for your

Reviewer: ```suggestion
consist of a root certificate authority (CA) and one or more intermediate CAs, which make up a trusted certificate chain for your
```

> Elasticsearch cluster. This is used to establish trust when Kibana creates an SSL connection with your Elasticsearch cluster. In addition to

Reviewer: ```suggestion
Elasticsearch cluster. This chain is used to establish trust when Kibana creates an SSL connection with your Elasticsearch cluster. In addition to
```

this setting, trusted certificates may be specified via `elasticsearch.ssl.certificateAuthorities` and/or `elasticsearch.ssl.keystore.path`.

`elasticsearch.ssl.truststore.password:`:: The password that will be used to decrypt the trust store. If your trust store has no password,
leave this unset. If your trust store has an empty password, set this to `""`.

`elasticsearch.ssl.verificationMode:`:: *Default: full* Controls the verification of certificates presented by Elasticsearch. Valid values

> are `none`, `certificate`, and `full`. `full` performs hostname verification, and `certificate` does not. This is only used when traffic to

Reviewer: ```suggestion
are `none`, `certificate`, and `full`. `full` performs hostname verification and `certificate` does not. This is only used when traffic to
```

> are `none`, `certificate`, and `full`. `full` performs hostname verification, and `certificate` does not. This is only used when traffic to

Reviewer: Qualifying "this":
```suggestion
are `none`, `certificate`, and `full`. `full` performs hostname verification, and `certificate` does not. This setting is used only when traffic to
```

> Elasticsearch is encrypted -- this is specified by using the HTTPS protocol in `elasticsearch.hosts`.

Reviewer: ```suggestion
Elasticsearch is encrypted, which is specified by using the HTTPS protocol in `elasticsearch.hosts`.
```

`server.socketTimeout:`:: *Default: "120000"* The number of milliseconds to wait before closing an
inactive socket.

`server.ssl.certificate:` and `server.ssl.key:`:: Paths to a PEM-encoded X.509 certificate and its private key, respectively. These are used
when enabling SSL for inbound requests from web browsers to the Kibana server.

> NOTE: These settings cannot be used in conjunction with `server.ssl.keystore.path`.

Reviewer: If this note should be indented under the server.ssl.certificate and server.ssl.key description:
```suggestion
+
--
NOTE: These settings cannot be used in conjunction with `server.ssl.keystore.path`.

--
```

> `server.ssl.certificateAuthorities:`:: Paths to one or more PEM-encoded X.509 certificates. This may consist of a root certificate authority

Reviewer: ```suggestion
`server.ssl.certificateAuthorities:`:: Paths to one or more PEM-encoded X.509 certificates. These certificates may consist of a root certificate authority
```

> (CA), and one or more intermediate CAs, which make up a trusted certificate chain for Kibana. This is used when a web browser creates an SSL

Reviewer: ```suggestion
(CA) and one or more intermediate CAs, which make up a trusted certificate chain for Kibana. This is used when a web browser creates an SSL
```

connection with the Kibana server; the certificate chain is sent to the browser along with the end-entity certificate to establish trust.

> This is also used to determine if client certificates should be trusted when PKI authentication is enabled. In addition to this setting,

Reviewer: Qualifying "this":
```suggestion
This setting is also used to determine whether client certificates should be trusted when PKI authentication is enabled. In addition to this setting,
```

Author: I meant "this chain" here, will qualify it accordingly.

> trusted certificates may be specified via `server.ssl.keystore.path` and/or `server.ssl.truststore.path`.

Author: The prior documentation for `server.ssl.certificateAuthorities` was not entirely accurate. This setting serves a dual purpose -- I don't really like it, but that's how it operates today. If we wanted to change this behavior and split it into two settings, we'd probably have to muck around Hapi and customize the verification code for client SSL certificates.

connections. Valid values are `required`, `optional`, and `none`. `required` forces a client to present a certificate, while `optional`
requests a client certificate but the client is not required to present one.

`server.ssl.enabled:`:: *Default: "false"* Enables SSL for inbound requests from the browser to the Kibana server. When set to `true`, a

> certificate and private key must be provided. These can be specified via `server.ssl.keystore.path`, or the combination of

Reviewer: ```suggestion
certificate and private key must be provided. These can be specified via `server.ssl.keystore.path` or the combination of
```

`server.ssl.certificate` and `server.ssl.key`.

`server.ssl.keyPassphrase:`:: The passphrase that will be used to decrypt the private key that is specified via `server.ssl.key`. This value
is optional, as the key may not be encrypted.

> `server.ssl.keystore.path:`:: Path to a PKCS #12 file -- this should contain an X.509 certificate with its private key. These are used when

Reviewer: ```suggestion
`server.ssl.keystore.path:`:: Path to a PKCS #12 file that contains an X.509 certificate with its private key. These are used when
```

enabling SSL for inbound requests from web browsers to the Kibana server. If the file contains any additional certificates, those will be

> used as a trusted certificate chain for Kibana. This is used when a web browser creates an SSL connection with the Kibana server; the

Reviewer: Qualifying "this":
```suggestion
used as a trusted certificate chain for Kibana. This chain is used when a web browser creates an SSL connection with the Kibana server; the
```

> certificate chain is sent to the browser along with the end-entity certificate to establish trust. This is also used to determine if client

Reviewer: ```suggestion
certificate chain is sent to the browser along with the end-entity certificate to establish trust. This setting is also used to determine whether client
```

Author: I meant "this chain" here, will qualify it accordingly.

certificates should be trusted when PKI authentication is enabled. In addition to this setting, trusted certificates may be specified via
`server.ssl.certificateAuthorities` and/or `server.ssl.truststore.path`.

> 

Reviewer: To indent the note under the setting's description:
```suggestion
+
--
```

> NOTE: This setting cannot be used in conjunction with `server.ssl.certificate` or `server.ssl.key`.

Reviewer: To end the indentation:
```suggestion
NOTE: This setting cannot be used in conjunction with `server.ssl.certificate` or `server.ssl.key`.

--
```

`server.ssl.keystore.password:`:: The password that will be used to decrypt the key store and its private key. If your key store has no
password, leave this unset. If your key store has an empty password, set this to `""`.

> `server.ssl.truststore.path:`:: Path to a PKCS #12 trust store -- this should contain one or more X.509 certificates. This may consist of a

Reviewer: ```suggestion
`server.ssl.truststore.path:`:: Path to a PKCS #12 trust store that contains one or more X.509 certificates. This may consist of a
```

> root certificate authority (CA), and one or more intermediate CAs, which make up a trusted certificate chain for Kibana. This is used when a

Reviewer: ```suggestion
root certificate authority (CA) and one or more intermediate CAs, which make up a trusted certificate chain for Kibana. This is used when a
```

web browser creates an SSL connection with the Kibana server; the certificate chain is sent to the browser along with the end-entity

> certificate to establish trust. This is also used to determine if client certificates should be trusted when PKI authentication is enabled.

Reviewer: ```suggestion
certificate to establish trust. This setting is also used to determine whether client certificates should be trusted when PKI authentication is enabled.
```

Author: I meant "this chain" here, will qualify it accordingly.
